### PR TITLE
Add github.com/syndtr/goleveldb/leveldb instrumentation

### PR DESCRIPTION
**Pull request by MrAlias**
This new module instruments the `github.com/syndtr/goleveldb/leveldb` package.

- [x] Comply with OpenTelemetry semantic conventions
- [x] Package documentation
- [x] Exported objects are documented
- [x] Tests and linting are passing
- [x] Includes dependabot config
- [x] Library added to README.md
- [x] CHANGELOG.md updated

### CODE REVIEW

File: instrumentation/github.com/syndtr/goleveldb/leveldb/splunkleveldb/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+## Compatibility
+
+- Go: 1.15+
+- github.com/syndtr/goleveldb/leveldb: 1.0.0+
+
+## Getting Started
+
+This package is designed to be used as a drop-in replacement for the use of the
+`github.com/syndtr/goleveldb/leveldb` package. Examples of how to do so can be
+found [here](./example_test.go).
```
Comment (tmorrisonsfx) [inline instrumentation/github.com/syndtr/goleveldb/leveldb/splunkleveldb/README.md:16]: Suggested change:  "Examples of how to do so can be
found [here](./example_test.go)." to "See example_test.go for more information."

**Review by tmorrisonsfx — APPROVED** [review]
One suggested change.

**Review by tmorrisonsfx — APPROVED** [review]

File: instrumentation/github.com/syndtr/goleveldb/leveldb/splunkleveldb/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Splunk instrumentation for `github.com/syndtr/goleveldb/leveldb`
+
+This package provides OpenTelemetry instrumentation for the
+[github.com/syndtr/goleveldb/leveldb](https://github.com/syndtr/goleveldb/leveldb)
+package.
+
+## Compatibility
+
+- Go: 1.15+
+- github.com/syndtr/goleveldb/leveldb: 1.0.0+
```
Comment (pellared) [inline instrumentation/github.com/syndtr/goleveldb/leveldb/splunkleveldb/README.md:10]: It is the first time I see such a section in this repository 😄 
Is it intended?

**Review by TomRoSystems — APPROVED** [review]

File: instrumentation/github.com/syndtr/goleveldb/leveldb/splunkleveldb/README.md (hunk 2 above)
Comment (MrAlias) [reply instrumentation/github.com/syndtr/goleveldb/leveldb/splunkleveldb/README.md:10]: It was intentional, but ultimately it is not needed. I'll remove it to remain consistent and progress the PR.

File: instrumentation/github.com/syndtr/goleveldb/leveldb/splunkleveldb/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
```
Comment (MrAlias) [inline instrumentation/github.com/syndtr/goleveldb/leveldb/splunkleveldb/README.md:11]:
```suggestion
```

**Pull request merged**